mapogcsld.c

```diff
@@ -3856,14 +3942,38 @@ char *msSLDGenerateTextSLD(classObj *psClass, layerObj *psLayer, int nVersion)
     strcpy(sNameSpace, "se:");
 
 
-  if (psClass && psLayer && psLayer->labelitem &&
-      strlen(psLayer->labelitem) > 0 && psClass->numlabels > 0) {
-    psLabelObj = psClass->labels[0];
+  if (!psLayer || !psClass) return pszSLD;
+
+  for (lid=0 ; lid < psClass->numlabels ; lid++)
+  {
+    char * psLabelText;
+    psLabelObj = psClass->labels[lid];
+
+    psLabelText = NULL;
+    if (psLabelObj->text.string)
+    {
+      psLabelText = msStrdup(psLabelObj->text.string);
```

Comment:
what if psLabelObj->text.string doesn't start with [ and ends with ] ? Should the type of the expression be checked ?

Reply:
You are correct. In my attempt to upgrade this part of code to MapServer version 6.2, I missed the whole picture and handled only few simple use cases. I'll rework on this as soon as I can. Up to you to decide whether the merge needs these upcoming changes or if it can be performed as is and changes be applied in another pull request.